### PR TITLE
Fix call to `wait` done post `SIGKILL`

### PR DESCRIPTION
**Pull request by future-highway**

### CODE REVIEW

**Comment by dmah42** [comment]
i think SIGINT is not appropriate as it is a "user interrupt request".  i would expect a parent to send SIGTERM to a child process as it gives an opportunity to clean up (unlike SIGKILL).  SIGQUIT is right out, given the core dump it generates.

have we considered SIGTERM?

**Comment by future-highway** [comment]
I think we should use whatever is most expected (i.e., standard practice). #199 seemed to indicate we want to use `SIGINT` but https://aurae.io/signals/ says to use `SIGTERM`.

**Comment by dmah42** [comment]
> I think we should use whatever is most expected (i.e., standard practice). #199 seemed to indicate we want to use `SIGINT` but https://aurae.io/signals/ says to use `SIGTERM`.

https://dsa.cs.tsinghua.edu.cn/oj/static/unix_signal.html <-- SIGTERM :)

File: auraed/src/runtime/cell_service/executables/auraed/nested_auraed.rs [hunk 1]
```diff
@@ -215,11 +215,14 @@ impl NestedAuraed {
         //       (contrary to ^^^). What do we want to signal?
 
         if wants_isolated_pid(&self.shared_namespaces) {
-            // TODO: Here, SIGINT works when using auraescript, but hangs(?) during unit tests.
-            //       SIGKILL, however, works in both scenarios. Why?
-            self.process.kill(Some(SIGKILL))?;
+            // TODO: Here, SIGTERM works when using auraescript, but hangs(?) during unit tests.
```
Comment (dmah42) [inline auraed/src/runtime/cell_service/executables/auraed/nested_auraed.rs:218]: sigterm hangs too? hm.

File: auraed/src/runtime/cell_service/executables/auraed/nested_auraed.rs [hunk 2]
```diff
@@ -215,11 +215,14 @@ impl NestedAuraed {
...
+            //       SIGKILL, however, works in both scenarios. This may be solved once we proxy
+            //       signals to nested auraeds.
+            self.process.kill(Some(SIGTERM))?;
             self.process.wait()
         } else {
+            // TODO: https://aurae.io/signals/ seems to indicate that we always want
+            //       to send SIGTERM to nested auraeds?
             self.process.kill(SIGKILL)?;
```
Comment (dmah42) [inline auraed/src/runtime/cell_service/executables/auraed/nested_auraed.rs:226]: i think SIGTERM too, but we should also handle that signal and shut down as cleanly as possible.

**Review by dmah42 — DISMISSED** [review]

**Review by dmah42 — APPROVED** [review]

**Pull request merged**